### PR TITLE
Drop deprecated crm_mon options

### PR DESCRIPTION
**Pull request by nrwahl2**
I stumbled across this task while dropping support for Nagios resources (the `--simple-status` machinery is for Nagios, but unrelated to resources).

Closes T707

### CODE REVIEW

File: tools/crm_mon.h [hunk 1]
```diff
@@ -46,7 +46,6 @@
 typedef enum mon_output_format_e {
     mon_output_unset,
     mon_output_none,
-    mon_output_monitor,
```
Comment (kgaillot) [inline tools/crm_mon.h:49]: update copyright

File: tools/crm_mon.c [hunk 2]
```diff
@@ -754,11 +747,6 @@ static GOptionEntry display_entries[] = {
 };
 
 static GOptionEntry deprecated_entries[] = {
-    { "as-xml", 'X', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, as_xml_cb,
```
[the remark this answers is not in the document]
Comment (nrwahl2) [reply tools/crm_mon.c:771]: Gross. Thank you for catching it early.

[the remark this answers is not in the document]
Comment (kgaillot) [reply tools/crm_mon.c:771]: FYI pgsql and redis check the feature set to determine which option to use. galera does so in some but not all calls.

**Review by kgaillot — APPROVED** [review]

File: tools/crm_mon.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply tools/crm_mon.c:771]: Dang, I should've done more than grep :)

**Comment by nrwahl2** [comment]
Rebased on current main to see if that will resolve the Coverity errors, since they refer to code that no longer exists.

**Pull request merged**